openmc/mesh.py

```diff
@@ -509,6 +510,49 @@ def from_rect_lattice(cls, lattice, division=1, mesh_id=None, name=''):
 
         return mesh
 
+    @classmethod
+    def from_bounding_box(
+        cls,
+        bounding_box,
```

Comment:
I'll suggest replacing 'bounding_box' with 'domain' throughout to capture what these incoming objects represent.

Reply:
ah yes that is a nice catch all term to use for cell, region, geometry, universe.